src/parser.ts

```diff
@@ -7015,7 +7015,13 @@ export function parseObjectLiteralOrPattern(
       let state = PropertyKind.None;
       let key: ESTree.Expression | null = null;
       let value;
-      if (parser.getToken() & Token.IsIdentifier || parser.getToken() === Token.EscapedReserved) {
+      if (
+        parser.getToken() & Token.IsIdentifier ||
+        parser.getToken() === Token.EscapedReserved ||
+        parser.getToken() === Token.EscapedFutureReserved
+      ) {
+        if (parser.getToken() === Token.EscapedFutureReserved) destructible |= DestructuringKind.CannotDestruct;
```

Comment:
Only in strict mode, our scanner emits Token.EscapedFutureReserved. So this added logic only affects strict mode.